File: galleries/movies/plotly_express/bubble_agg.py
import pandas as pd
import plotly.express as px

df = pd.read_csv("data/movies.csv")
df2 = df.groupby("MPAA Rating").mean(numeric_only=True).reset_index()
fig = px.scatter(
    df2,
    x="Production Budget",
    y="Worldwide Gross",
    color="MPAA Rating",
    size="IMDB Rating",
)
fig

File: galleries/movies/ggplot2/bubble_agg.R
library(tidyverse)

df <- read_csv("data/movies.csv") |>
  group_by(`MPAA Rating`) |>
  summarize_all(~ mean(.x, na.rm = TRUE))

ggplot(df) +
  geom_point(aes(
    x = `Production Budget`, y = `Worldwide Gross`,
    color = `MPAA Rating`, size = `IMDB Rating`
  ))
